### PR TITLE
Small fixes to linkerd2 chart README.md

### LINKED ISSUE #5092: controllerLogLevel - Chart values incorrectly documenated
https://github.com/linkerd/linkerd2/blob/827646a3e107fc1b048dc8a8f830f640270ba7f6/charts/linkerd2/templates/controller.yaml#L79

The chart readme values table shows `controllerLogLevel` while it actually under global at usage.

### PR DESCRIPTION
**Pull request by alpeb**
Fixes #5092: a couple of entries should go under `global` and fixed a typo.

### CODE REVIEW

**Review by kleimkuhler — APPROVED** [review]

File: charts/linkerd2/README.md [hunk 1]
```diff
@@ -102,6 +100,8 @@ their default values.
 | `global.clusterDomain`                      | Kubernetes DNS Domain name to use                                                                                                                                                     | `cluster.local`                      |
 | `global.cniEnabled`                         | Omit the NET_ADMIN capability in the PSP and the proxy-init container when injecting the proxy; requires the linkerd-cni plugin to already be installed                               | `false`                              |
 | `global.controllerComponentLabel`           | Control plane label. Do not edit                                                                                                                                                      | `linkerd.io/control-plane-component` |
+| `global.controllerImageVersion`             | Tag for the controller container docker image                                                                                                                                         | latest version                       |
```
Comment (adleong) [inline charts/linkerd2/README.md:103]: controllerImageVersion and global.controllerImageVersion both exist.  I don't understand what the difference between them is or why they both exist though.

Comment (alpeb) [reply charts/linkerd2/README.md:103]: @adleong where do you see `controllerImageVersion` outside the `global` context?

Comment (adleong) [reply charts/linkerd2/README.md:103]: https://github.com/linkerd/linkerd2/blob/main/pkg/charts/linkerd2/values.go#L26

Comment (alpeb) [reply charts/linkerd2/README.md:103]: Ok I think the one without `global.` is some leftover from a previous refactoring, as it's not used anywhere in the linkerd2 chart templates. If so, we should remove it from values.go, options.go and some other places.

**Review by adleong — APPROVED** [review]

**Pull request merged**